src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,305 @@
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @DeleteMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}/deregister/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> deregisterStudent(@PathVariable Long courseId, @PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to deregister {} student from tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> studentToDeregister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (studentToDeregister.isEmpty()) {
+            throw new EntityNotFoundException("User", studentLogin);
+        }
+        tutorialGroupService.deregisterStudent(studentToDeregister.get(), tutorialGroupFromDatabase);
+        return ResponseEntity.ok().body(null);
+    }
+
+    /**
+     * POST courses/:courseId/tutorial-groups/:tutorialGroupId/register/:studentLogin : register a student to a tutorial group.
+     *
+     * @param courseId        the id of the course to which the tutorial groups belong to
+     * @param tutorialGroupId the id of the tutorial group
+     * @param studentLogin    the login of the student to register
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @PostMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}/register/{studentLogin:" + Constants.LOGIN_REGEX + "}")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Void> registerStudent(@PathVariable Long courseId, @PathVariable Long tutorialGroupId, @PathVariable String studentLogin) {
+        log.debug("REST request to register {} student to tutorial group : {}", studentLogin, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdWithTeachingAssistantAndRegistrationsElseThrow(tutorialGroupId);
+        checkTutorialCourseIdMatchesPathId(courseId, tutorialGroupFromDatabase);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+
+        Optional<User> studentToRegister = userRepository.findOneWithGroupsAndAuthoritiesByLogin(studentLogin);
+        if (studentToRegister.isEmpty()) {
+            throw new EntityNotFoundException("User", studentLogin);
+        }
+        tutorialGroupService.registerStudent(studentToRegister.get(), tutorialGroupFromDatabase);
+        return ResponseEntity.ok().body(null);
```

Comment:
`ResponseEntity.noContent()` might also make sense here.

Reply:
Done in 94342a84726518898c07e0ced71a30fc0c478410